### PR TITLE
Extend QueryProps from QueryOpts

### PR DESCRIPTION
**Pull request by nikita-graf**
`QueryOpts` interface is currently missing a property `partialRefetch` which is present in `QueryProps`. It prevents using `partialRefetch` in a `query` HOC. Rather than adding this property to `QueryOpts` I'm using extending. Hope it'll help to prevent such a diff between two interfaces in the future.

### Checklist:

* [ ] If this PR is a new feature, please reference an issue where a consensus about the design was reached (not necessary for small changes)
* [ ] Make sure all of the significant new logic is covered by tests
* [ ] If this was a change that affects the external API used in GitHunt-React, update GitHunt-React and post a link to the PR in the discussion.

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -42,9 +42,10 @@ export interface QueryOpts<TGraphQLVariables = OperationVariables> {
   fetchPolicy?: FetchPolicy;
   errorPolicy?: ErrorPolicy;
   pollInterval?: number;
-  client?: ApolloClient<any>;
+  client?: ApolloClient<Object>;
```
Comment (rosskevin) [inline src/types.ts]: I don't believe `ApolloClient<Object>` to be correct.  What is the rationale for changing it?

Comment (nikita-graf) [reply src/types.ts]: As far as `QueryProps` interface defines `client` as [`ApolloClient<Object>`](https://github.com/apollographql/react-apollo/blob/master/src/Query.tsx#L95) I preserved it within my changes. I agree that `any` would be a better option for both interfaces.

Comment (rosskevin) [reply src/types.ts]: In this file you have changed it so I would change it back.  I'm pretty certain `Object` is incorrect.

Comment (nikita-graf) [reply src/types.ts]: Reverted back `any` type.

**Comment by rosskevin** [comment]
Thanks @nikita-graf

**Pull request merged**